udata/features/territories/models.py

```diff
@@ -24,6 +27,10 @@ def slug(self):
         return '{territory_id}-{id}'.format(
             territory_id=self.territory.id.replace('/', '-'), id=self.id)
 
+    @property
+    def organization(self):
+        return Organization.objects.get(id=self.organization_id)
```

Comment:
Would mongoengine cache this query or would each call to `territory_dataset.organization` make a new request?